server/lib/s3backup/__init__.py

```diff
@@ -246,6 +267,30 @@ def __init__(self, config):
                                          aws_secret_access_key='{}'.format(
                                              self.secret_access_key),
                                          endpoint_url='{}'.format(self.endpoint_url))
+        self.logger = logger
+
+    @staticmethod
+    def calculate_multipart_etag(tb, chunk_size):
+        md5s = []
+
+        with open(tb, 'rb') as fp:
+            while True:
```

Comment:
Why not have a for loop?

```
for data in fp.read(chunk_size):
    md5s.append(hashlib.md5(data))
```

Reply:
Uhh - let the record show an embarassed silence...